simulator/src/scenarios/key_rotation.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{
+	constants::{
+		self, PRIVATE_FOLLOW_CONNECTIONS, PRIVATE_FRIENDSHIP_CONNECTIONS,
+		PRIVATE_FRIENDSHIP_PAGE_MODIFICATIONS,
+	},
+	scenarios::common::{add_keys_for_users, compare_on_chain_with_expected, modify_random_pages},
+	GlobalState,
+};
+use dsnp_graph_config::Environment;
+use dsnp_graph_core::api::api_types::*;
+use rand::{prelude::SliceRandom, thread_rng};
+
+pub fn execute_key_rotation_private_follow(state: &mut GlobalState, env: Environment) {
```

Comment:
Hmm, this makes me think... I know we talked about a future enhancement to support a key rotation top-level method in the API, but I don't think we ever created an issue for it...

Reply:
We have `force_recalculate_graphs` that does key rotation on all pages also but currently we only do key rotation lazily whenever a page is modified.